packages/trader/src/index.tsx

```diff
@@ -1,27 +1,53 @@
 import React from 'react';
 import { getPositionsV2TabIndexFromURL, isDTraderV2, makeLazyLoader, moduleLoader, routes } from '@deriv/shared';
 import { Loading } from '@deriv/components';
+import { useGrowthbookGetFeatureValue } from '@deriv/hooks';
+import { TCoreStores } from '@deriv/stores/types';
+import { TWebSocket } from 'Types';
+import { Analytics } from '@deriv-com/analytics';
 
-const App = makeLazyLoader(
-    () =>
-        moduleLoader(() => {
-            if (isDTraderV2()) {
-                return import(/* webpackChunkName: "trader-app-v2", webpackPreload: true */ './AppV2/index');
-            }
-            return import(/* webpackChunkName: "trader-app", webpackPreload: true */ './App/index');
-        }),
-    () => {
-        if (isDTraderV2())
-            return (
-                <Loading.DTraderV2
-                    initial_app_loading
-                    is_contract_details={window.location.pathname.startsWith('/contract/')}
-                    is_positions={window.location.pathname === routes.trader_positions}
-                    is_closed_tab={getPositionsV2TabIndexFromURL() === 1}
-                />
-            );
-        return <Loading />;
-    }
-)();
+type Apptypes = {
+    passthrough: {
+        root_store: TCoreStores;
+        WS: TWebSocket;
+    };
+};
+
+const AppLoader = makeLazyLoader(
+    () => moduleLoader(() => import(/* webpackChunkName: "trader-app", webpackPreload: true */ './App/index')),
+    () => <Loading />
+)() as React.ComponentType<Apptypes>;
+
+const AppV2Loader = makeLazyLoader(
+    () => moduleLoader(() => import(/* webpackChunkName: "trader-app-v2", webpackPreload: true */ './AppV2/index')),
+    () => (
+        <Loading.DTraderV2
+            initial_app_loading
+            is_contract_details={window.location.pathname.startsWith('/contract/')}
+            is_positions={window.location.pathname === routes.trader_positions}
+            is_closed_tab={getPositionsV2TabIndexFromURL() === 1}
+        />
+    )
+)() as React.ComponentType<Apptypes>;
 
+const App = ({ passthrough }: Apptypes) => {
+    const [dtrader_v2_enabled] = useGrowthbookGetFeatureValue({
```

Comment:
Will new value from `useGrowthbookGetFeatureValue` hook triggers App component re-rendering? In case of 'yes' do we need then part with setting state in useEffect?